add api changes for 2.x (#588)
SAP/cloud-sdk: docs/js/features/odata/common/entity/middleware.mdx

> In earlier versions of the SAP Cloud SDK, `moment` was the default type for all date/time properties in the OData clients. But `moment` is now a legacy project and in maintenance mode. It also doesn't work well with modern "tree shaking" algorithms, leading to a large application bundle size.

Reviewer: ```suggestion
In earlier versions of the SAP Cloud SDK, `moment` was the default type for all date/time properties in the OData clients. 
But `moment` is now a legacy project and in maintenance mode. 
It also doesn't work well with modern "tree shaking" algorithms, leading to a large application bundle size.
```

> Therefore, in version 2.0 of the SDK, the default (de)serialization for date/time fields has been changed from `moment` to [`Temporal`](https://tc39.es/proposal-temporal/docs/). Also, a new flexible middleware concept has been introduced that allows configuring custom implementations for serialization and de-serialization of all types. You can set your custom middleware by passing it as a parameter to the entity's `Service()` method. This parameter is optional. When left unset, the default implementation is used.

Reviewer: Maybe I have missed it, but I remember there was a discussion what should be the default for 2.0, Temporal or Moment. Are you sure we decided to use Temporal?

```suggestion
Therefore, in version 2.0 of the SAP Cloud SDK, the default (de)serialization for date/time fields has been changed from `moment` to [`Temporal`](https://tc39.es/proposal-temporal/docs/). 
Also, a new flexible middleware concept has been introduced that allows configuring custom implementations for serialization and de-serialization of all types. 
You can set your custom middleware by passing it as a parameter to the entity's `Service()` method. 
This parameter is optional. 
When left unset, the default implementation is used.
```

```ts
const customMiddleware = {
  'Edm.DateTime': {
    deserialize: dayjsCustomDeserializationFunc
    serialize: dayjsCustomSerializationFunc
  }
}

const { businessPartnerApi } = businessPartnerService(customMiddleware);
```

> #### Using default `Temporal` Middleware

Reviewer: ```suggestion
#### Using Default `Temporal` Middleware
```

> `Temporal` proposes classes for all datatypes required to handle OData types (for example, `PlainDate`, `PlainTime`, `PlainDateTime`, `Duration` and, `ZonedDateTime`). The example below shows how to set a value of an `Edm.DateTime` field.

Reviewer: ```suggestion
`Temporal` proposes classes for all data types required to handle OData types (for example, `PlainDate`, `PlainTime`, `PlainDateTime`, `Duration` and, `ZonedDateTime`). 
The example below shows how to set a value of an `Edm.DateTime` field.
```

```ts
const { businessPartnerApi } = businessPartnerService();
businessPartnerApi
  .entityBuilder()
  .organizationFoundationDate(
    Temporal.PlainDateTime.from('1995-12-07T03:24:30')
  )
  .build();
```

#### Setting `moment` as Middleware

> We will continue to make `moment` based middleware available for cases where a switch to `Temporal` is not possible. This implementation is available as a separate npm package `@sap-cloud-sdk/moment-middleware`. Adapt your code to use this package:

Reviewer: ```suggestion
We will continue to make `moment`-based middleware available for cases where a switch to `Temporal` is impossible. 
This implementation is available as a separate npm package `@sap-cloud-sdk/moment-middleware`. 
Adapt your code to use this package:
```
